src/templates/csr/fabric-video/index.svelte

```diff
@@ -0,0 +1,219 @@
+
+	export let BackgroundScrollType: GAMVariable<'parallax' | 'none' | 'fixed'>;
+
+	export let Layer1BackgroundImage: GAMVariable;
+	export let Layer1BackgroundPosition: GAMVariable;
+	export let Layer2BackgroundImage: GAMVariable;
+	export let Layer2BackgroundPosition: GAMVariable;
+	export let Layer3BackgroundImage: GAMVariable;
+	export let Layer3BackgroundPosition: GAMVariable;
+	export let MobileLayer1BackgroundImage: GAMVariable;
+	export let MobileLayer1BackgroundPosition: GAMVariable;
+	export let MobileLayer2BackgroundImage: GAMVariable;
+	export let MobileLayer2BackgroundPosition: GAMVariable;
+	export let MobileLayer3BackgroundImage: GAMVariable;
+	export let MobileLayer3BackgroundPosition: GAMVariable;
+	export let VideoURL: GAMVariable;
+	export let VideoBackupImage: GAMVariable;
+	export let MobileVideoBackupImage: GAMVariable;
+	export let VideoURLMobile: GAMVariable;
+	export let VideoAlignment: GAMVariable;
+	export let thirdPartyJSTracking: GAMVariable;
+	export let isXL = false;
+
+	const isMobile = window.innerWidth < 740;
+
+	const video: Writable<HTMLVideoElement | undefined> = writable();
+
+	let played = false;
+
+	// Add video source
+	// Add video poster image
+	const posterImage = isMobile ? MobileVideoBackupImage : VideoBackupImage;
+	const videoSrc = isMobile ? VideoURLMobile : VideoURL;
+
+	video.subscribe((video) => {
+		if (video) {
+			if (!VideoURL || !VideoURLMobile) return;
+
+			video.load(); //
+			void video.play(); //
+
+			const observer = new IntersectionObserver(
+				(entries) => {
+					entries.forEach((entry) => {
+						if (entry.isIntersecting && !played) {
+							video.paused && void video.play();
+						} else {
+							!video.paused && video.pause();
+						}
+					});
+				},
+				{ root: null, rootMargin: '0px', threshold: 0.2 },
+			);
+			observer.observe(video);
+		}
+	});
+</script>
+
+<a
+	class="fabric-container"
+	class:is-parallax={BackgroundScrollType === 'parallax'}
+	class:is-xl={isXL}
+	href={`${CLICK_MACRO}${DEST_URL}`}
+	target="_blank"
+>
+	<div
+		class="layer"
+		style:--desktop-background-image={`url('${Layer1BackgroundImage}')`}
+		style:--desktop-background-position={Layer1BackgroundPosition}
+		style:--mobile-background-image={`url('${MobileLayer1BackgroundImage}')`}
+		style:--mobile-background-position={MobileLayer1BackgroundPosition}
+	/>
+	<div
+		class="layer"
+		style:--desktop-background-image={`url('${Layer2BackgroundImage}')`}
+		style:--desktop-background-position={Layer2BackgroundPosition}
+		style:--mobile-background-image={`url('${MobileLayer2BackgroundImage}')`}
+		style:--mobile-background-position={MobileLayer2BackgroundPosition}
+	/>
+	<div
+		class="layer"
+		style:--desktop-background-image={`url('${Layer3BackgroundImage}')`}
+		style:--desktop-background-position={Layer3BackgroundPosition}
+		style:--mobile-background-image={`url('${MobileLayer3BackgroundImage}')`}
+		style:--mobile-background-position={MobileLayer3BackgroundPosition}
```

Comment:
There's lots of duplicate code from the Fabric component - at least 30 lines if not more - could we use that here and add the video as an option inside the component? Feels like it would be easier to maintain if we used the shared component!

Reply:
@emma-imber  👍 Refactored to a more generic Fabric Component